code/game/objects/structures/displaycase.dm

```diff
@@ -1,79 +1,116 @@
 /obj/structure/displaycase
 	name = "display case"
-	icon = 'icons/obj/stationobjs.dmi'
-	icon_state = "glassbox1"
 	desc = "A display case for prized possessions. It taunts you to kick it."
-	density = 1
-	anchored = 1
-	unacidable = 1//Dissolving the case would also delete the gun.
+	icon = 'icons/obj/display_case.dmi'
+	icon_state = "glassbox"
+	density = TRUE
+	anchored = TRUE
+	unacidable = TRUE
+	req_access = list(access_captain)
 	var/health = 30
-	var/occupied = 1
-	var/destroyed = 0
+	var/obj/held_obj
+	var/open = FALSE
+	var/destroyed = FALSE
+	var/spawn_contained_type
+
+/obj/structure/displaycase/Initialize(mapload)
+	. = ..()
+	if(spawn_contained_type)
+		var/obj/O = new spawn_contained_type(src)
+		held_obj = O
+		update_icon()
 
 /obj/structure/displaycase/ex_act(severity)
 	switch(severity)
 		if (1)
-			new /obj/item/material/shard( src.loc )
-			if (occupied)
-				new /obj/item/gun/energy/captain( src.loc )
-				occupied = 0
-			qdel(src)
+			do_destroy()
 		if (2)
-			if (prob(50))
-				src.health -= 15
-				src.healthcheck()
+			if(prob(50))
+				health -= 15
+				health_check()
```

Comment:
Just check the health outside the switch instead of putting it in two different cases.